### PR TITLE
Use *bool for MaintainerCanModify 

### PR DESCRIPTION
**Pull request by msal4**
#### Summary
Decouple model from database-specific fields

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16975

### CODE REVIEW

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Looks good @msal4 But could you please exclude `.idea/` contents? You can add these to project `.gitignore`.

**Review by mahmud2011 — CHANGES_REQUESTED** [review]
You forgetfully committed `.idea/`

**Comment by msal4** [comment]
Oops! you never saw that

File: util/ptr.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package util
+
+func Boolptr(b bool) *bool { return &b }
```
Comment (agnivade) [inline util/ptr.go:3]: Please move this inside the server package as an unexported function.

**Review by agnivade — CHANGES_REQUESTED** [review]

File: util/ptr.go (hunk 1 above)
Comment (msal4) [reply util/ptr.go:3]: I found out that these helper functions already exist in `utils`. I'm going to use them instead.

**Review by agnivade — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]
Thanks @msal4 🎉

**Pull request merged**